callback/authorize.php

```diff
@@ -69,15 +70,15 @@ function getAccessToken($authorization_code) {
     }
 	$_SESSION["access_token"]=json_decode($tokenResult)->access_token;
 	$_SESSION["refresh_token"]=json_decode($tokenResult)->refresh_token;
+	$_SESSION["type"] = htmlspecialchars($_GET["type"]);
```

Comment:
Why not just $type? You already have it gone through htmlspecialchars

Reply:
using $type caused errors by passing empty type